crates/prover-services/src/parallel/prover.rs

```diff
@@ -93,13 +96,26 @@ where
         })
     }
 
-    pub(crate) fn submit_witness(
+    pub(crate) fn submit_assumptions(
+        &self,
+        assumptions: Vec<Vec<u8>>,
+        da_slot_hash: <Da::Spec as DaSpec>::SlotHash,
+    ) {
+        let header_hash = da_slot_hash;
+        let mut prover_state = self.prover_state.lock();
+        let prover_status = prover_state.prover_status.get_mut(&header_hash);
+        if let Some(ProverStatus::WitnessSubmitted(data)) = prover_status {
+            data.1 = assumptions
+        }
+    }
+
+    pub(crate) fn submit_input(
```

Comment:
I feel like these 2 are dangerous, as they can overwrite each other. I think behavior for these 2 methods should go like:
1. lock
2. if entry doesnt exist, create a default empty entry.
3. only update the assumption or input based on whatever is the method.

Maybe we don't use it in that manner at the current state in the code, but they shouldn't be able to overwrite each other imo. Specifically, submit_input -> submit_assumptions.

Reply:
This was also in my mind, thanks for the heads up